src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Orders/DataBoxEdgeOrderGetCmdlet.cs

```diff
@@ -0,0 +1,103 @@
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.EdgeGateway;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using System.Collections.Generic;
+using System.Management.Automation;
+using PSResourceModel = Microsoft.Azure.PowerShell.Cmdlets.DataBoxEdge.Models.PSDataBoxEdgeOrder;
+using PSTopLevelResource = Microsoft.Azure.PowerShell.Cmdlets.DataBoxEdge.Models.PSDataBoxEdgeDevice;
+using ResourceModel = Microsoft.Azure.Management.EdgeGateway.Models.Order;
+
+
+namespace Microsoft.Azure.PowerShell.Cmdlets.DataBoxEdge.Common.Cmdlets.Orders
+{
+    [Cmdlet(VerbsCommon.Get, Constants.Order, DefaultParameterSetName = GetByNameParameterSet
+     ),
+     OutputType(typeof(PSResourceModel))
+    ]
+    public class DataBoxEdgeOrderGetCmdlet : AzureDataBoxEdgeCmdletBase
+    {
+        private const string GetByResourceIdParameterSet = "GetByResourceIdParameterSet";
+        private const string GetByDeviceObjectParameterSet = "GetByDeviceObjectParameterSet";
+        private const string GetByNameParameterSet = "GetByNameParameterSet";
+
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = GetByDeviceObjectParameterSet,
+            HelpMessage = Constants.PsDeviceObjectHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public PSTopLevelResource DeviceObject { get; set; }
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = GetByResourceIdParameterSet,
+            HelpMessage = Constants.ResourceIdHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = GetByNameParameterSet,
+            HelpMessage = Constants.ResourceGroupNameHelpMessage,
+            Position = 0)]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = GetByNameParameterSet,
+            HelpMessage = Constants.ResourceGroupNameHelpMessage,
+            Position = 1)]
+        [ResourceNameCompleter("Microsoft.DataBoxEdge/dataBoxEdgeDevices", nameof(ResourceGroupName))]
+        [ValidateNotNullOrEmpty]
+        public string DeviceName { get; set; }
```

Comment:
ValueFromPipelineByPropertyName = true

Reply:
Please use Name as parameter name, and use DeviceName as alias. Same for other cmdlet.
Normally we prefer to use Name for current level resource.

Reply:
For each and every Order's resource name is constant and value of order's name is _default_ and is under parent resource DataBoxEdge's resource which is referred here as DeviceName.
Since taking Name as parameter is redundant skipped accepting Name as parameter but in the future for some reason name can be reused

Reply:
In this case, please keep DeviceName